### ISSUE TITLE
Hi, we are preparing backdoor-attack works for CVPR2022. Could you please provide us the training codes?

### ISSUE THREAD

**Issue by ddghjikle** [yuezunli/ISSBA#1]
Hi, thanks very much for sharing the wonderful work and its codes. It is no doubt that this work provides insights of the backdoor-attack area. Recently, we are preparing new works for CVPR2022. In these new works, we want to compare its training processes with that of your work. Therefore, we hope you can provide complete codes such as training parts. Then, we will be able to systematically compare your work with ours and comprehensively discuss advantages of your work.

**Comment by yuezunli** [comment]
Hi, thanks for your interest in this work. We will update the code with the training process ASAP. We will let you know once it is done.

**Comment by ddghjikle** [comment]
Hi, it is glad to see several training codes are updated. However, it seems that existing training processes are different from the paper. For example, the model dose not contain a decoder. During training, the encoder does not take target labels as inputs. Similarly, the Code Reconstruction Loss and Image Reconstruction Loss in Figure 3 are not calculated.

**Comment by zeabin** [comment]
@ddghjikle  Hi, I am reading this paper too. If I am not mistaken, the training of encoder and the training of backdoor attack are seperated. And the code released at present do not contain the process for training encoder and generating trigger images using encoder. The losses you metioned are the losses for encoder, thus not calculated here. Since the encoder used is [StegaStamp](https://github.com/tancik/StegaStamp), you can find the code in that repository.

**Comment by ddghjikle** [comment]
Hi,&nbsp;it&nbsp;is&nbsp;true&nbsp;your&nbsp;explaination&nbsp;is&nbsp;helpful&nbsp;for&nbsp;understanding&nbsp;this&nbsp;wonderful&nbsp;work.&nbsp;However,&nbsp;if&nbsp;the&nbsp;encoder&nbsp;and&nbsp;decoder&nbsp;are&nbsp;seperately&nbsp;trained,&nbsp;then&nbsp;the&nbsp;two&nbsp;losses&nbsp;in&nbsp;Figure&nbsp;3&nbsp;are&nbsp;difficult&nbsp;to&nbsp;compute.&nbsp;To&nbsp;be&nbsp;specific,&nbsp;from&nbsp;the&nbsp;above&nbsp;figure,&nbsp;it&nbsp;can&nbsp;be&nbsp;seen&nbsp;that&nbsp;the&nbsp;given&nbsp;target&nbsp;label&nbsp;is&nbsp;coded&nbsp;and&nbsp;compared&nbsp;with&nbsp;the&nbsp;decoded&nbsp;target&nbsp;label,&nbsp;which&nbsp;is&nbsp;generated&nbsp;by&nbsp;the&nbsp;Decoded&nbsp;Code.&nbsp;Therefore,&nbsp;it&nbsp;can&nbsp;be&nbsp;concluded&nbsp;that&nbsp;the&nbsp;encoder&nbsp;and&nbsp;decoder&nbsp;should&nbsp;be&nbsp;jointly&nbsp;trained&nbsp;at&nbsp;least&nbsp;once&nbsp;in&nbsp;each&nbsp;iteration.&nbsp;Of&nbsp;course,&nbsp;thanks&nbsp;very&nbsp;muck&nbsp;for&nbsp;pointing&nbsp;out&nbsp;the&nbsp;StegaStamp&nbsp;encoder.&nbsp;However,&nbsp;it&nbsp;is&nbsp;easy&nbsp;to&nbsp;find&nbsp;that&nbsp;the&nbsp;encoder&nbsp;defined&nbsp;in&nbsp;the&nbsp;provided&nbsp;codes&nbsp;takes&nbsp;two&nbsp;images&nbsp;(one&nbsp;benign&nbsp;and&nbsp;one&nbsp;poisoned)&nbsp;as&nbsp;inputs,&nbsp;rather&nbsp;than&nbsp;coded&nbsp;Target&nbsp;Label&nbsp;and&nbsp;Benign&nbsp;Image.
------&nbsp;原始邮件&nbsp;------
***@***.***&gt;;
发送时间:2021年9月14日(星期二)&nbsp;晚上11:45
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:Re:&nbsp;[yuezunli/ISSBA]&nbsp;Hi,&nbsp;we&nbsp;are&nbsp;preparing&nbsp;backdoor-attack&nbsp;works&nbsp;for&nbsp;CVPR2022.&nbsp;Could&nbsp;you&nbsp;please&nbsp;provide&nbsp;us&nbsp;the&nbsp;training&nbsp;codes?&nbsp;(#1)



 
@ddghjikle  Hi, I am reading this paper too. If I am not mistaken, the training of encoder and the training of backdoor attack are seperated. And the code released at present do not contain the process for training encoder and generating trigger images using encoder. The losses you metioned are the losses for encoder, thus not calculated here. Since the encoder used is StegaStamp, you can find the code in that repository.
 
&mdash;
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by zeabin** [comment]
Hi, I mean the encoder and decoder are jointly trained, but seperate from training stage of backdoor attack. The workflow is:
1. train encoder and decoder jointly (Figure 5)
2. generate poisoned images using the encoder (Figure 4(a) Attack Stage)
3. embed a backdoor using the poisoned and benign images (Figure 4(b) Training Stage)

And the encoder in `encode_image.py` takes `secret` and `image` as inputs, not two images.

**Comment by v-mipeng** [comment]
Will you upload the code for the whole process, including the generation of poisoned samples and model training on the poisoned data set.

**Comment by THUYimingLi** [comment]
> Will you upload the code for the whole process, including the generation of poisoned samples and model training on the poisoned data set.

Hi, thanks for your attention to our work. You can use our [toolbox](https://github.com/THUYimingLi/BackdoorBox) to run ISSBA directly. Please find more details from our [codes](https://github.com/THUYimingLi/BackdoorBox/blob/main/core/attacks/ISSBA.py) and [test example](https://github.com/THUYimingLi/BackdoorBox/blob/main/tests/test_ISSBA.py).
